index.rst

```diff
@@ -1,61 +1,263 @@
-..
-  Technote content.
+:tocdepth: 1
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+Abstract
+========
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+This technote gathers information about the history, current state and contemplates future evolutions of the Vera Rubin Observatory Control System (Rubin-OCS) Middleware.
+The middleware is the backbone of the Rubin-OCS.
+The highly distributed nature of the Rubin-OCS places tight constraints in terms of latency, availability and reliability for the middleware.
+Here we gather information to answer some common questions regarding technology choices, describe some of the in-house work done to obtain a stable system and document some of the concerns with the current state of the system, potential impacts for near-future/commissioning and future/operations.
+We also cover some of the work we have been doing to investigate alternative technologies and suggest some potential roadmaps for the future of the system.
 
-  Use the following syntax for sections:
+Introduction
+============
 
-  Sections
-  ========
+"Middleware" is the term used to describe software applications used for communication in sofware systems.
+These technologies became popular with the advent of *distributed systems*, wich come as a solution to the problem of parallel computation.
+As software and hardware systems became more and more complex, it become impractical to develop and execute them in a single process and, eventually, a single node.
+With that, software evolved from *monolith* applications, where a single program executes in a single process or node, to *distributed* applications, where the system is divided into a number of smaller applications; each running on their own process or node.
+For these applications to work together coherently they must be able to comunicate with each other, thus giving origin to middleware technologies.
 
-  and
+How a *distributed system* is broken down into smaller pieces is heavily dependent uppon the problem.
+Some systems are only broken down into a small number of components each still in charge of large contexts, others are broken down into several small applications that are in charge only of small simple tasks.
+The later has gained substantial popularity recently and is commonly referred to as *microservices*.
+These systems are behind many of popular large services in use today like Google and Amazon. 
 
-  Subsections
-  -----------
+The architecture of *distributed systems* can take many shapes and forms.
+For instance, some systems are designed to mimick or emulate *monolith* applications.
```

Comment:
"mimic" and "monolithic"

But "mimic or emulate" is redundant and I think "emulate" is the better verb, so I suggest "...are designed to emulate monolithic..."

Reply:
I have a few concerns about this section and I'm not sure entirely what to suggest:
* It seems a bit long and detailed
* It starts out saying there are many approaches, but only details two. It might be clearer to say "two popular options are..."

I am uneasy about the terms  "monolithic" and "event-driven reactive". Those names have no obvious relationship to each other; it is not even clear to me that are mutually exclusive. Certainly monolithic even-driven systems are quite common -- all GUI-based applications fit that category.

It feels to me as if you are really detailing the difference between hierarchical and peer-to-peer systems (there may be better terms for this). I'll have to read on to see if this how you refer back to this information later. At the moment I have no idea where you are going with it.

Certainly DDS is obviously peer-to-peer. But the way we intend to use Kafka is, as well. The main difference between them is that one is distributed and the other is broker-based. I think any message-based system is going to be tend towards peer-to-peer (or at least a shallow hierarchy) due to the difficulty of pushing messages down through many layers.